pkg/kubectl/cmd/delete.go

```diff
@@ -95,8 +96,37 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		return err
 	}
 
+	// By default use a reaper to delete all related resources.
+	if cmdutil.GetFlagBool(cmd, "recursive") {
+		return ReapResult(r, f, out)
+	}
+	return DeleteResult(r, out)
+}
+
+func ReapResult(r *resource.Result, f *cmdutil.Factory, out io.Writer) error {
+	found := 0
+	err := r.Visit(func(info *resource.Info) error {
```

Comment:
Looks like `DeleteResult` is ignoring `NotFound` errors. Should this do the same?  I noticed that some`Reapers` are aggressively getting objects before deletion.